src/component.ts

```diff
@@ -64,5 +64,10 @@ export function componentFactory (
   const Super = superProto instanceof Vue
     ? superProto.constructor as VueClass
     : Vue
-  return Super.extend(options)
+  const rv = Super.extend(options);
+
+  for(let sttc in Component)
+    if(!(sttc in function() {}) && Component.hasOwnProperty(sttc))
```

Comment:
> `!(sttc in function() {})`

Is this necessary? Looks like only using `hasOwnProperty` is enough.

Reply:
This will for instance avoid statics to override ` name`  or ` apply`

Reply:
Why shouldn't they be overridden? Is that stated in the spec?
I think we should follow the es spec as possible since the user would expect it.